src/service/resources-impl.js

```diff
@@ -834,16 +834,46 @@ export class ResourcesImpl {
         ) {
           // 7. The new height (or one of the margins) is smaller than the
           // current one.
-        } else if (
-          request.newHeight == box.height &&
-          this.isWidthOnlyReflowFreeExpansion_(
-            resource.element,
-            request.newWidth || 0
-          )
-        ) {
-          // 8. Element is in viewport, but this is a width-only expansion that
-          // should not cause reflow.
-          resize = true;
+        } else if (request.newHeight == box.height) {
+          // 8. Element is in viewport, but this is a width-only expansion.
+          // Check whether this should be reflow-free, in which case,
+          // schedule a size change.
+          this.vsync_.run(
+            {
+              measure: state => {
+                const parent = resource.element.parentElement;
+                if (!parent) {
+                  state.resize = false;
+                  return;
+                }
+
+                // If the element has siblings, it's possible that a width-expansion will
+                // cause some of them to be pushed down.
+                const parentWidth =
+                  (parent.getLayoutWidth && parent.getLayoutWidth()) ||
+                  parent./*OK*/ offsetWidth;
+                let cumulativeWidth = widthDiff;
+                for (let i = 0; i < parent.childElementCount; i++) {
+                  cumulativeWidth += parent.children[i]./*OK*/ offsetWidth;
+                  if (cumulativeWidth > parentWidth) {
+                    state.resize = false;
+                    return;
+                  }
+                }
+                state.resize = true;
+              },
+              mutate: state => {
+                if (state.resize) {
```

Comment:
Should `overflowCallback` be called?

Reply:
If I'm reading the code right, I don't think it's relevant in this case as there's no overflow element.

Reply:
Discussed with Dima; realized callback is necessary & added.